openstack/messaging/v2/queues/requests.go

```diff
@@ -148,3 +148,12 @@ func Get(client *gophercloud.ServiceClient, queueName string, clientID string) (
 		MoreHeaders: map[string]string{"Client-ID": clientID}})
 	return
 }
+
+// Delete deletes the specified queue.
+func Delete(client *gophercloud.ServiceClient, queueName string, clientID string) (r DeleteResult) {
+	_, r.Err = client.Request("DELETE", deleteURL(client, queueName), &gophercloud.RequestOpts{
```

Comment:
Quick note on this: it should be possible to do `client.Delete` instead of `client.Request` and still pass the `MoreHeaders` option. But let me know if that didn't work for you.